Documentation/network/clustermesh/mcsapi.rst

```diff
@@ -0,0 +1,165 @@
+      kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/mcs-api/4690877cc89bcfd9302d516ae43515f446d0aecf/config/crd/multicluster.x-k8s.io_serviceimports.yaml
+
+
+To install Cilium with MCS-API support, run:
+
+   .. code-block:: shell-session
+
+      helm install cilium |CHART_RELEASE| \\
+      --namespace kube-system \\
+      --set clustermesh.enableMCSAPISupport=true
+
+To enable MCS-API API support on an existing Cilium installation, run:
+
+   .. code-block:: shell-session
+
+      helm upgrade cilium |CHART_RELEASE| \\
+      --namespace kube-system \\
+      --reuse-values \\
+      --set clustermesh.enableMCSAPISupport=true
+      kubectl -n kube-system rollout restart deployment/cilium-operator
+      kubectl -n kube-system rollout restart deployment/clustermesh-apiserver
+
+Also checkout the `EndpointSlice synchronization <endointslicesync>`_ feature
+if you need Headless Services support.
+
+Installing CoreDNS multicluster
+-------------------------------
+
+You also need to install and configure the `multicluster CoreDNS plugin <multicluster_coredns>`_.
+
+.. _multicluster_coredns: https://github.com/kubernetes/enhancements/blob/master/keps/sig-multicluster/1645-multi-cluster-services-api/README.md
+
+The multicluster CoreDNS plugin is currently not provided by default in
+CoreDNS so you will have to recompile it yourself.
+
+To rebuild the image you need to add the multicluster plugin to plugins.cfg
+file. The ordering of plugins matters, add it just below kubernetes plugin
+that has very similar functionality:
+
+   .. parsed-literal::
+
+      ...
+      kubernetes:kubernetes
+      multicluster:github.com/coredns/multicluster
+      ...
+
+Then you can build your image simply by running ``make`` and then ``docker build .``
+with the right docker tag and upload it to your preferred registry.
+
+Also you will to make sure that CoreDNS is able to read and watch the relevant
+MCS-API resources (ServiceImports). You can do so by running the following
+command on each cluster:
+
+   .. code-block:: shell-session
+
+      kubectl patch clusterrole system:coredns --type=json \
+         -p='[{"op":"add","path":"/rules/-","value":{"apiGroups":["multicluster.x-k8s.io"],"resources":["serviceimports"],"verbs":["list","watch"]}}]'
+
+After that you will need to update the CoreDNS's corefile to also enable the
+multicluster plugin, for instance by executing the following command on each cluster:
+
+   .. code-block:: shell-session
+
+      kubectl get configmap -n kube-system coredns -o yaml | \
+         sed -E -e 's/^(\s*)kubernetes.*cluster\.local.*$/\1multicluster clusterset.local\n&/' | \
+         kubectl replace -f-
+
+And you can finally deploy the CoreDNS image you previously built. Doing so will
+also rollout the CoreDNS deployment and activate the corefile change you previously made.
```

Comment:
FYI I hope to be able to simplify this process in the near(-ish) future

Reply:
That would be awesome :rocket: